### ISSUE TITLE
doc: docker install procedure installs insecure worker

### ISSUE THREAD

**Issue by rgerhards** [buildbot/buildbot#4179]
The docker latent worker doc states that ```pip install buildbot-worker``` shall be used to install the worker: http://docs.buildbot.net/latest/manual/cfg-workers-docker.html#id5

However, this does not install the docker worker that understands the ```WORKER_ENVIRONMENT_BLACKLIST``` environment variable. The  ```worker.tac``` file is quite different from this version here: https://github.com/buildbot/buildbot/blob/master/worker/docker/buildbot.tac

In the end result, sensitive information is given inside the log files. I have seen that in your Dockerfile you "manually" copy over the tac file. I think this is not really optimal, but if required should be mentioned in the doc.

... but I am not sure if I overlook the obvious. Is there another pip package that I can use to install the docker worker?

**Comment by tardyp** [comment]
Indeed, that page was written before the official Dockerfile and not updated since. Would you like to update it to rather mention https://github.com/buildbot/buildbot/blob/master/worker/Dockerfile as the official example?

**Comment by rgerhards** [comment]
Thanks for the answer!

> Indeed, that page was written before the official Dockerfile and not updated since. Would you like to update it to rather mention https://github.com/buildbot/buildbot/blob/master/worker/Dockerfile as the official example?

I can see that I do that, but I am not sure what the correct fix is. Is it to say "You must use the official Dockerfile?" (note that I am not too buildbot-literate). Or do you mean one should clone the buildbot repo and the place one's own Docker definitions into it - that would IMHO be an even worse solution.

In any case, a doc update is not a solution to my concern: What do I need to install if I do not want to start out from your official Dockerfile, but from another one. I have containers for our testbench environment, and they do the heavy hauling. We also plan to extend them to several base distros (replacing current VMs), so it is not an option to base on a single official buildbot container. Right now,  I do the dirty workaround of overwriting buildbot.tac, but I really do not consider this a valid solution.

concrete: https://github.com/rsyslog/rsyslog-docker/blob/master/dev_env/ubuntu/buildbot/16.04/Dockerfile

Do I misunderstand something on the workflow?

**Comment by tardyp** [comment]
you would have to copy the buildbot.tac around your dockerfile, and then apply the install procedure as found in the official dockerfile.

That buildbot.tac is pretty docker specific, this is why this is not part of the official buildbot.tac generated by buildbot create worker. Perhaps we should eventually add a flag do create-worker

**Comment by rgerhards** [comment]
> you would have to copy the buildbot.tac around your dockerfile, and then apply the install procedure as found in the official dockerfile.

That's actually what I am doing. It works...

> That buildbot.tac is pretty docker specific, this is why this is not part of the official buildbot.tac generated by buildbot create worker. Perhaps we should eventually add a flag do create-worker

... but that means I now need to manually ensure that it works with what pip installs. A flag to create-worker would be great to have!

**Comment by rgerhards** [comment]
I think I am now ready to help update the doc ... but I do not find where. Probably I overlooked something. Could you point me to it?
